### DISCUSSION TITLE
Set icon colour by sensor state

### DISCUSSION THREAD

**Discussion by Boredmanny** [Blackymas/NSPanel_HA_Blueprint#1750] [Q&A] [answered]
Hi all,

I’ve got a sensor (sensor.next_waste_collection) that tells me what the next waste collection is (Blue or green top bin). I’d like to be able to have an icon (mid:trash) on the Home Screen that is green or blue tied to this sensor. I’ve got the icon as one of the custom buttons but how do I set its colour? It feels like this should be doable in the blueprint yaml but I’m struggling to find the documentation for adjusting the yaml, probably because the UI has been the focus. Can anyone help?

**Comment by edwardtfn** [comment]
Similar to:
- #1183 
- #1736

**Reply by Boredmanny** [reply]
Thank you, I’m struggling with what the specific text should be, I tried this but it didn’t work, I’m using this as a way to teach myself code so apologies if I’m doing anything daft!:
‘’’
    home_custom_button07_icon_color: >-
        {% if is_state("sensor.next_waste_collection", "Refuse Bin for Non-Recycling") -%}
          "Green"
        {%- else -%}
          "Blue"
        {%- endif %}
‘’’

**Reply by Boredmanny** [reply]
And sorry I’m also learning how to share code snippets properly!

**Comment by Boredmanny** [comment]
```
home_custom_button07_icon_color: >-
  {% if is_state("sensor.next_waste_collection", "Refuse Bin for Non-Recycling") -%}
     "Green"
  {%- else -%}
     "Blue"
  {%- endif %}
```

**Reply by edwardtfn** [reply]
I'm not sure if this will work, but if it works, it should be like this:
```yaml
home_custom_button07_icon_color: >-
  {% if is_state("sensor.next_waste_collection", "Refuse Bin for Non-Recycling") -%}
     [0, 255, 0]
  {%- else -%}
     [0, 0, 255]
  {%- endif %}
```
Or this:

```yaml
home_custom_button07_icon_color: >
  {{
    [0, 255, 0]
    if is_state("sensor.next_waste_collection", "Refuse Bin for Non-Recycling")
    else [0, 0, 255]
  }}
```

Or this:

```yaml
home_custom_button07_icon_color: '{{ [0, 255, 0] if is_state("sensor.next_waste_collection", "Refuse Bin for Non-Recycling") else [0, 0, 255] }}'
```

This is the RGB (Red, Green, Blue) code for those colors.

**Reply by edwardtfn** [reply]
Please let us know if this works.

**Reply by Boredmanny** [reply]
Away with work for a couple of days but will try at the weekend, thanks for the fast feedback.

**Reply by Boredmanny** [reply]
I tried the first version you suggested and it didn’t work. I was then thinking, keep it simple and just try to change the colour so I tried: 
```

home_custom_button07_icon_color: >-
     [0, 255, 0]

```

which it updated on save to:

```
home_custom_button07_icon_color: "[0, 255, 0]"
```

but this didn’t work. Is ‘home_custom_button07_icon_color’ the right line?

**Reply by edwardtfn** [reply]
If you are editing `automations.yaml`, or the automation in yaml mode, then the input id should be `home_custom_button07_icon_color`, but looks unlikely that this will support templating.

The format for your example should be
```yaml
home_custom_button07_icon_color: [0, 255, 0]
```
Which is exactly the same as:
```yaml
home_custom_button07_icon_color:
  - 0
  - 255
  - 0
```
And is exactly the same as manually selecting green in the color picker using the UI mode.

**Reply by Boredmanny** [reply]
Thanks for the help. Yes I’m editing the yams in the automations UI, click the three dots top right and switch to yaml. So are you saying that should work, do the quotation marks make any difference?  I started because I’m sure I saw a picture in a features section that said you could individually change all the colours of icons and showed that with every icon a different colour. Do you know if there’s another way to achieve the same result. I was thinking maybe make an on / off template sensor from the original sensor with on being green bin, off blue and then I could maybe use a chip or possibly two sensors and two chips (one blue, one green)?

**Reply by Boredmanny** [reply]
https://user-images.githubusercontent.com/41958506/226168254-e3d7be56-f2ca-4830-b1b2-96025ecb2e4e.png

This picture

**Reply by edwardtfn** [reply]
Look, you can freely set the color of the icons and there's no need to go to yaml for that. But that is static. If on chips, the icon will be shown with the selected color or hidden, for buttons, the icon will be shown if the off color or the selected color (if any), and for other entities, will always show with the selected color.
However, what you asked on this thread was a conditional color based on state. Then that is a different thing and you can alternatively use scripts/automations in HA as suggested by @kroessig at https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1183#discussioncomment-8209579.

**Reply by Boredmanny** [reply]
Understood, and sorry if testing patience. The end result I am looking for is a quick clean indicator of what the next bin collection is, a green or blue trash icon. 

On the UI you can’t set the colour of the home icons directly that I can see, you can for chips or the entity pages but not home page icons, I may have missed something though? If I could that was going to be a fault finding step, eg change it in the UI and then adjust to include templating in the yaml. 

The reason I was using yaml to make it go green was fault finding why I couldn’t change the colour at all, then once I got that working it’s a smaller step to see if I can get the templating working. It maybe that templating doesn’t work and I have to find another way to achieve the aim or wait until templating is supported.

**Reply by edwardtfn** [reply]
No need to apologies. We are all learning here. And trying to have fun. 😉 

For a notification for trash/recycling bin, probably the easiest thing is to use chips, so you set a binary sensor in Home Assistant to be on when it's time to put the bin out, assign that to a chip with an icon and a color you like and it will be shown in the top of the panel when it is `on`. You can even use multiple chips for each of the bins, with a different color (and icon) for each one.
If you need help to create those binary sensors in Home Assistant, please let us know (and provide some info about the sensor you have).

You can, of course, use other sensors, if you wanna have something with more focus. There are 3 values shown in the home page and those will support text, so you can assign your sensor directly to those and the text will be there, with the color you chose.
These values, however, are not conditional. They will be shown always with the same color and will always be visible (unless your sensor provides an empty string).

**Comment by Boredmanny** [comment, marked as answer]
Working :)

Made two opposite template binary sensors using the original sensor and then used these to drive two chip icons, one blue and one green. Did think that I could have done it with one binary sensor and then had two chips, one of which driven by the binary sensor but inverted, however working now so will leave as is.

Thanks for the help, I’m still getting my head around templating but good to experiment and try and thanks for your patience.
